app.js

```diff
@@ -60,15 +60,51 @@ app.post('/api/addUserRequest', function (req, res) {
 
   // process entry
   if (checkParams.valid === true) {
-    var pendingRequest = req.body;
-    pendingRequest["time_submitted"] = new Date();
-    pendingRequest["helped"] = false;
 
-    db.child("requests").push(req.body, function () {
-      res.status(200).end();
+    var hasDuplicateCase = false;
+
+    // check if a request with the same contact info has not already been submitted
+    db.child("cases").once("value", function (snapshot) {
+      if (snapshot.val() !== null) {
+        for (var k in snapshot.val()) {
+          var numberPattern = /\d+/g;
+
+          if (req.body.contactMethod == "SMS") {
```

Comment:
No, that's more of a bad habit on my part. I guess we should keep things strict. Thanks for feedback.